pkgs/by-name/xn/xnconvert/package.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchurl,
+  appimageTools,
+  makeDesktopItem,
+  imagemagick,
+  runCommand,
+}:
+
+let
+  icon =
+    runCommand "xnconvert-icon.png"
+      {
+        nativeBuildInputs = [ imagemagick ];
+        src = fetchurl {
+          url = "https://www.xnview.com/img/app-xnconvert-512.webp";
+          hash = "sha256-le+rvthQndY3KbkPYuMGZDDcvdpvH9CIS2REP1vmDXg=";
+        };
+      }
+      ''
+        convert $src $out
+      '';
+  desktopItem = (
+    makeDesktopItem {
+      name = "xnconvert";
+      desktopName = "XnConvert";
+      exec = "xnconvert";
+      icon = "xnconvert";
+      comment = "A fast, powerful and free cross-platform batch image converter.";
+      categories = [ "Graphics" ];
+    }
+  );
+in
+appimageTools.wrapType2 rec {
+  pname = "xnconvert";
+  version = "1.105.0";
```

Comment:
Not blocking (purely a style choice), but you may move the version to the let-in block to remove the `rec` entirely! (Fell free to mark it has resolved)